nomad/structs/host_volumes.go

```diff
@@ -180,18 +180,68 @@ const (
 
 // HostVolumeStub is used for responses for the list volumes endpoint
 type HostVolumeStub struct {
-	Namespace string
-	ID        string
-	Name      string
-	PluginID  string
-	NodePool  string
-	NodeID    string
-	Capacity  int64 // bytes
-	State     HostVolumeState
+	Namespace     string
+	ID            string
+	Name          string
+	PluginID      string
+	NodePool      string
+	NodeID        string
+	CapacityBytes int64
+	State         HostVolumeState
 
 	CreateIndex uint64
 	CreateTime  int64
 
 	ModifyIndex uint64
 	ModifyTime  int64
 }
+
+type HostVolumeCreateRequest struct {
+	Volumes []*HostVolume
+	WriteRequest
+}
+
+type HostVolumeCreateResponse struct {
+	Volumes []*HostVolume
+	WriteMeta
+}
+
+type HostVolumeRegisterRequest struct {
+	Volumes []*HostVolume
+	WriteRequest
+}
+
+type HostVolumeRegisterResponse struct {
+	Volumes []*HostVolume
+	WriteMeta
+}
+
+type HostVolumeDeleteRequest struct {
+	VolumeIDs []string
+	WriteRequest
+}
+
+type HostVolumeDeleteResponse struct {
+	WriteMeta
```

Comment:
partial deletes are possible, right? so maybe returning deleted volume IDs would be useful

```suggestion
type HostVolumeDeleteResponse struct {
	VolumeIDs []string
	WriteMeta
```

I suspect this will have implications where this struct is used

Reply:
Good call. In practice I wasn't able to make the HTTP API nice to use with multiple volumes, because of the existing URL routes for CSI which we're sharing. So being able to delete multiple volumes is going to be primarily only here to support soft deletes and periodic GC, which will be in later PRs.